beef_tx_sorting.go

```diff
@@ -34,11 +36,11 @@ func prepareSortStructures(dag []*Transaction) (txByID map[string]*Transaction,
 	return
 }
 
-func calculateIncomingEdges(inDegree map[string]int, txByID map[string]*Transaction) {
+func calculateIncomingEdges(inDegree map[string]int, txByID map[string]*bt.Tx) {
 	for _, tx := range txByID {
-		for _, input := range tx.draftTransaction.Configuration.Inputs {
-			inputUtxoTxID := input.UtxoPointer.TransactionID
-			if _, ok := txByID[inputUtxoTxID]; ok { // transaction can contains inputs we are not interested in
+		for _, input := range tx.Inputs {
+			inputUtxoTxID := input.PreviousTxIDStr() // TODO: perf
```

Comment:
On what measures we assume that performance optimisation is needed?

Reply:
I added a more detailed comment